packages/lib/services/rest/routes/notes.test.ts

```diff
@@ -78,4 +78,44 @@ describe('routes/notes', () => {
 
 		expect(response).toBe('');
 	});
+
+	test('should not copy content from invalid protocls', async () => {
+		const url = 'file:///home/user/file.db';
+		const shimFsDriverCopySpy = jest.fn();
+		jest.spyOn(shim, 'fsDriver').mockImplementation(() => {
+			return {
+				copy: shimFsDriverCopySpy,
+			} as any;
+		});
+
+		const allowedProtocols: string[] = [];
+		await downloadMediaFile(url, null, allowedProtocols);
+
+		expect(shimFsDriverCopySpy).toBeCalledTimes(0);
+	});
+
+	test.each([
+		'https://joplinapp.org/valid/image_url',
+		'https://joplinapp.org/valid/image_url.invalid_url',
+	])('should find and move file with invalid or without filename', async (url) => {
+		jest.spyOn(shim, 'fetchBlob').mockImplementation(() => {
+			return {
+				headers: {
+					'content-type': 'image/jpg',
+				},
+			};
+		});
```

Comment:
I guess it makes sense to mock `fetchBlob` to avoid http requests in tests, but for the other mocks I'm not sure, I think they can be avoided. It's nice to know in advance the UUID but again if we use another uuid lib, this test will be broken. Maybe you can check what's in the directory before and after to find what you need.

Reply:
This is trickier to do without spyOn. 

What I'm testing is when the image type can't be found by the URL. 

We fetch the image and save it on filesystem with `fetchBlob` function, since we don't want a HTTP request in the test I'm using a mock for it, so no file gets created. Then the code will call `tryToGuessExtFromMimeType`, since the URLs that I used for the test don't have an extension, which will try to move the file that was never created (that is why I have a mock for `shim.fsDriver.move`)

In the end we can't even look at the temp directory to see if the test worked because no files has ever been created in this test. Maybe it would be best to remove this test altogether?